packages/runtime/container-runtime/src/test/containerRuntime.spec.ts

```diff
@@ -2644,6 +2661,82 @@ describe("Runtime", () => {
 					"SignalLost and SignalOutOfOrder telemetry should not be logged when signals are processed in order",
 				);
 			});
+
+			it("increments signalLost field after successive dropped signals", () => {
+				// Send 5 signals
+				sendSignals(5);
+
+				// Process the first signal
+				processSignals(submittedSignals, 1);
+
+				// Drop the second signal
+				dropSignals(1);
+
+				// Process the third signal
+				processSignals(submittedSignals, 1);
+
+				// Missing second signal should be detected
+				logger.assertMatch(
+					[
+						{
+							eventName: "ContainerRuntime:SignalLost",
+							signalsLost: 1,
+						},
+					],
+					"SignalLost telemetry should be logged when signal is dropped",
+				);
+
+				// Drop the fourth signal
+				dropSignals(1);
+
+				// Process the fifth signal
+				processSignals(submittedSignals, 1);
+
+				// Missing fourth signal should be detected
+				logger.assertMatch(
+					[
+						{
+							eventName: "ContainerRuntime:SignalLost",
+							signalsLost: 2,
+						},
+					],
+					"SignalLost telemetry should be logged when signal is dropped",
+				);
+			});
+
+			it("ignores in-flight signals on disconnect/reconnect", () => {
+				// Send 5 signals
+				sendSignals(4);
+
+				// Process one signal
+				processSignals(submittedSignals, 1);
+
+				// Disconnect + Reconnect
+				changeConnectionState(containerRuntime, false, mockClientId);
+				changeConnectionState(containerRuntime, true, mockClientId);
+
+				// Temporarily lose two old signals
+				dropSignals(2);
+
+				// Receive one old signal sent before disconnect
+				processSignals(submittedSignals, 1);
```

Comment:
Does this come across as local or remote? In a real scenario, will it be local or remote?

Reply:
It comes across as local, and in a real scenario it will be local.